src/App.tsx

```diff
@@ -1,8 +1,85 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+import { getUsers, getTodos, Todo } from './helpers/api';
+import { TodoList } from './components/TodoList';
+import { SortButtons } from './components/SortButtons';
+
+const App = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [sortBy, setSortBy] = useState(0);
+
+  const handleLoadClick = async () => {
+    setIsLoading(true);
+
+    const todosFromServer = await getTodos();
+    const usersFromServer = await getUsers();
+
+    const todosWithUsers = todosFromServer.map(todo => ({
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId),
+    }));
+
+    setInitialTodos(todosWithUsers);
+  };
+
+  const reset = () => {
+    setSortBy(0);
+  };
+
+  const sortByUserName = () => {
+    setSortBy(1);
+  };
+
+  const sortByTitle = () => {
+    setSortBy(2);
+  };
+
+  const sortByCompleted = () => {
+    setSortBy(3);
+  };
+
+  const todos = [...initialTodos].sort((a, b) => {
```

Comment:
try useMemo for this case so that your array will recalculate only if initialTodos are changed